### PR TITLE
docs: api reference

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Add API reference page.

- Generate the static html file of API Reference according to the specification `https://logto.dev/api/swagger.json`
    - Should put the HTML file in the directory `/static/*` for storing static resources that need to be controlled by Docusaurus.
        - [How to link a static HTML file not generated by Docusaurus · Issue #3894 · facebook/docusaurus](https://github.com/facebook/docusaurus/issues/3894)
- Add the link to the static html file: [Navbar link](https://docusaurus.io/docs/api/themes/configuration#navbar-link)

References:

- Redoc: 
    - Deploy via CLI: https://redocly.com/docs/redoc/deployment/cli/#redoc-cli-commands
    - `redoc-cli build` options: https://github.com/Redocly/redoc/blob/2100b549c0f1d54c95c41fd0d3eb0b6b23deccb1/cli/index.ts#L68
- Docusaurus: 
    - Navbar: https://docusaurus.io/docs/api/themes/configuration#navbar-link
    - How to link a static HTML file not generated by Docusaurus · Issue #3894 · facebook/docusaurus: https://github.com/facebook/docusaurus/issues/3894

## Linear Issue Reference

LOG-2609
LOG-2812

## Testing

- Navbar link to API Reference

    image

- Click the link and open a new Logto Core API Reference page

    image

FIXME:

- [x] peer dependencies from `redoc-cli`
    image
- [x] ~unexpected wepack failure warning~

### CODE REVIEW

File: packages/docs/docusaurus.config.js [hunk 1]
```diff
@@ -64,6 +64,12 @@ const config = {
             position: 'left',
             label: 'SDK',
           },
+          {
+            href: '/api',
+            target: '_blank',
```
Comment (wangsijie) [inline packages/docs/docusaurus.config.js:69]: is this by design?

**Review by wangsijie — APPROVED** [review]

File: packages/docs/docusaurus.config.js (hunk 1 above)
Comment (IceHe) [reply packages/docs/docusaurus.config.js:69]: Yes, added a link to the Redoc page (static html file).

See Slack thread [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1654334346129139?thread_ts=1654331790.882009&cid=C02A8G4HVAM)

File: packages/docs/.gitignore [hunk 2]
```diff
@@ -7,6 +7,7 @@
 # Generated files
 .docusaurus
 .cache-loader
+/static/api/index.html
```
Comment (demonzoo) [inline packages/docs/.gitignore:10]: Is this html auto generated?

Comment (IceHe) [reply packages/docs/.gitignore:10]: Yes. Generated by `redoc-cli build https://logto.dev/api/swagger.json -o static/api/index.html --title 'Logto Core API Reference'`

---

Reference: see the PR summary

- Generate the static html file of API Reference according to the specification https://logto.dev/api/swagger.json
    - Should put the HTML file in the directory /static/* for storing static resources that need to be controlled by Docusaurus.
        - https://github.com/facebook/docusaurus/issues/3894

Comment (demonzoo) [reply packages/docs/.gitignore:10]: ok, I see

**Review by demonzoo — APPROVED** [review]

**Comment by IceHe** [comment]
Rebased master to resolve conflicts.

**Pull request merged**